source/material_model/rheology/elasticity.cc

```diff
@@ -298,61 +354,166 @@ namespace aspect
+        // MaterialModelInputs are based on 'solution' when calling the MaterialModel for the reaction rates.
+        // This means that we can use 'in' to get to the $\tau^{0}$ and velocity/strain rate of the
+        // previous timestep.
+        if (in.current_cell.state() == IteratorState::valid && this->get_timestep_number() > 0)
+          {
+            for (unsigned int i = 0; i < in.n_evaluation_points(); ++i)
+              {
+                // Set all reaction rates to zero
+                for (unsigned int c = 0; c < in.composition[i].size(); ++c)
+                  reaction_rate_out->reaction_rates[i][c] = 0.0;
+
+                // Get $\tau^{0}$ of the previous timestep t from the compositional fields.
+                // This stress includes the rotation and advection.
+                SymmetricTensor<2, dim> stress_0_t;
+                for (unsigned int j = 0; j < SymmetricTensor<2, dim>::n_independent_components; ++j)
+                  stress_0_t[SymmetricTensor<2, dim>::unrolled_to_component_indices(j)] = in.composition[i][j];
+
+                // $\eta^{t}_{effcreep}
+                const double effective_creep_viscosity = out.viscosities[i];
+
+                // $\eta_{el} = G \Delta t_{el}$
+                const double elastic_viscosity = calculate_elastic_viscosity(average_elastic_shear_moduli[i]);
+
+                // The total stress of timestep t.
+                // TODO: Do we somehow need to take into account the user-set minimum value for the
+                // sqrt of the second invariant of the strain rate?
+                const SymmetricTensor<2, dim>
+                stress_t = 2. * effective_creep_viscosity * (deviator(in.strain_rate[i]) + stress_0_t / (2. * elastic_viscosity));
+
+                // Fill reaction rates.
+                // Assume dte is always equal to dt.
+                // During this timestep, the reaction rates will be multiplied
+                // with the current timestep size to turn the rate of change into a change.
+                // However, this update belongs
+                // to the previous timestep. Therefore we divide by the
+                // current timestep and multiply with the previous one.
+                // When multiplied with the current timestep, this will give
+                // (rate * compensation_factor) * current_dt = (rate * previous_dt / current_dt) * current_dt = previous_change.
+                // However, we need to compute a rate, and therefore divide
+                // the change in stress by the previous timestep:
+                // rate = \Delta_stress / previous_dt.
+                // Therefore, the only thing we need to do is to divide by the
+                // current timestep.
+                //const double previous_dt = this->get_old_timestep();
+                const double current_dt  = this->get_timestep();
+                //const double timestep_compensation_factor = previous_dt / current_dt;
```

Comment:
Can tis line, line 506 (`//const double previous_dt = this->get_old_timestep();`) be removed since they are longer used? Likewise, should the documentation above be updated to just include lines 501-505?